### PR TITLE
Simplify-routes

### PR DESCRIPTION
**Pull request by editid0**
- Add 404 page
- Add used packages/libraries to readme
- Made routes use the existing dict to reduce duplication
- Updated all pages with analytics
- Updates some pages to use the container box layout
- Formatted pages with prettier
- Made tools on the path /tools/toolname
- Use .env to update the AI limit per ip per day

## Summary by Sourcery

Simplify and enhance the routing system by using a dictionary to reduce duplication, introduce a 404 error page, and update pages with analytics. Improve the layout by applying a container box design to some pages and format HTML with Prettier. Update the README to include a list of used packages and libraries.

New Features:
- Introduce a 404 error page to handle non-existent routes.

Enhancements:
- Refactor routes to use a dictionary for reduced duplication and improved maintainability.
- Update all pages to include analytics tracking.
- Apply a container box layout to some pages for a more consistent design.
- Format HTML pages using Prettier for improved code readability.

Documentation:
- Add a list of used packages and libraries to the README for better documentation.

### CODE REVIEW

**Comment by editid0** [comment]
@sourcery-ai review

File: main.py [hunk 1]
```diff
@@ -239,8 +227,8 @@ def colorai():
         data[today_string][ip_hash] = 1
     with open("data.json", "w") as f:
         json.dump(data, f, indent=4)
-    # check if the count is greater than 10
-    if data[today_string][ip_hash] > 10:
+    # check if the count is greater than DAILY_AI_LIMIT
```
[the remark this answers is not in the document]
Comment (editid0) [reply main.py:230]: still in development, so it's a bit impractical to use a database for now

File: main.py [hunk 2]
```diff
@@ -59,150 +58,157 @@ class Color(BaseModel):
     )
 app = Flask(__name__)
 
+@app.errorhandler(404)
+def page_not_found(e):
+    return render_template("404.html"), 404
 
 our_tools = [
```
[the remark this answers is not in the document]
Comment (editid0) [reply main.py:66]: Why is it better to have each route seperate?

[the remark this answers is not in the document]
Comment (editid0) [reply main.py:66]: would it be better or worse to have them in a seperate file?

**Review by mahadali05 — APPROVED** [review]

**Pull request merged**